### PR TITLE
Fix bug where filename was not being updated when we follow a referen…

### LINKED ISSUE #208: resolve refs incorrectly resolves files with conflicting namespaces in its definitions dict
If I have this swagger spec
'''
{
    "swagger": "2.0",
    "info": {
        "title": " ",
        "version": "0.0.0"
    },
    "paths": {
        "/foo": {
            "get": {
                "responses": {
                    "200": {
                        "description": " ",
                        "schema": {
                            "$ref": "foo.json#/common"
                        }
                    }
                }
            }
        },
        "/bar": {
            "get": {
                "responses": {
                    "200": {
                        "description": " ",
                        "schema": {
                            "$ref": "bar.json#/common"
                        }
                    }
                }
            }
        }
    }
}
'''
There "common" references in two different files. Lets assume foo.json looks like this
'''
{
    "common": {
        "$ref": "#definitions/common"
    },
    "definitions": {
        "common": {
            "type": "integer"
        }
    }
}
'''
And bar.json looks like this:
'''
{
    "common": {
        "$ref": "#definitions/common"
    },
    "definitions": {
        "common": {
            "type": "string"
        }
    }
}
'''
When I call api.resolve_refs on my spec, my defs dict will look something like this:
'''
{
        ":....bar.json|..common": {
            "$ref": "#/definitions/:....bar.json|definitions..common"
        },
        ":....foo.json|..common": {
            "$ref": "#/definitions/:....foo.json|definitions..common"
        },
        ":....|definitions..common": {
            "type": "integer"
        }
    }
'''
As you can see, the definition in bar.json for common being a string has been overwritten.

### PR DESCRIPTION
**Pull request by shrayus**
resolve #208

### CODE REVIEW

File: tests/api_test.py [hunk 1]
```diff
@@ -112,3 +112,18 @@ def test_extenal_refs_no_empty_keys():
     spec = Spec.from_dict(spec_dict, path)
     flattened_spec = resolve_refs(spec, spec_dict, ['/'], 'swagger', {})
     traverse_spec(flattened_spec)
+
+
+def test_extenal_conflicting_namespaces():
```
Comment (jnb) [inline tests/api_test.py]: Did you check that this test fails without your fix?

File: tests/api_test.py [hunk 2]
```diff
@@ -112,3 +112,18 @@ def test_extenal_refs_no_empty_keys():
...
+    """
+    This test ensures that we dont overwrite
```
Comment (jnb) [inline tests/api_test.py]: Nit: I think you can probably use longer lines here?

File: tests/api_test.py (hunk 1 above)
Comment (shrayus) [inline tests/api_test.py]: yup! It fails, never been this proud of a failing test ;)

**Comment by striglia** [comment]
@shrayus is this still relevant or will we fix this w/ #216?

**Comment by macisamuele** [comment]
@striglia this PR is relevant to flattening code that has been removed by #214 .

**Comment by striglia** [comment]
Cool @macisamuele, thought it might be outdated now. I'll close this out then, thanks!

**Pull request closed without merging**